app/models/workers_scaling/algorithms/sample_algorithm.rb

```diff
@@ -160,9 +157,19 @@ def decrease_computational_power
     # If average_throughput is 0, throughput_needed value is ignored and exactly one Worker is added
     # Returns predicted throughput growth
     def add_workers(infrastructure, average_throughput = 0, throughput_needed = 0)
-      workers_needed = average_throughput > 0 ? (throughput_needed / average_throughput).ceil : 1
-      LOGGER.debug "Starting #{workers_needed} Workers on infrastructure: #{infrastructure}"
-      @resources_interface.schedule_workers(workers_needed, infrastructure).count * average_throughput
+      workers_needed = if average_throughput > 0
```

Comment:
To co napisałem nie działa, nieskończoność się łapie za wcześnie, zostaw tak jak jest.